### PR TITLE
qt: Fix minimized window bug on Linux

### LINKED ISSUE #14591: Minimized window bug on Linux
If the main window has been minimized to the taskbar (iconified) calling "Show/Hide", "Send" or "Receive" actions from the tray icon menu has no effect: the main window still minimized.

It seems like a Qt bug similar to that old [one](https://bugreports.qt.io/browse/QTBUG-31117).

Tested platform: Linux Mint 19, Qt 5.9.5 / 5.9.6

### PR DESCRIPTION
**Pull request by hebasto**
Fix #14591 

On some Linux systems the minimized to the taskbar (iconified) main window cannot be restored properly using actions from the systray icon menu when `QSystemTrayIcon::contextMenu()` is a child of the main window.

### CODE REVIEW

**Comment by fanquake** [comment]
Given that this is touching the same function, and fixing similar behaviour to #14222, I'd combine this into that PR. Please keep the two commits seperate though, if these fixing different bugs etc.

**Comment by hebasto** [comment]
@fanquake it seems #14123 fixes the same issue as #14222 and the latter can be closed, if the former is merged. Do you insist on combining this PR into #14222 ?

**Comment by hebasto** [comment]
Rebased after #14123 merging.

@promag @jonasschnelli 
Would you mind reviewing this PR?

File: src/qt/guiutil.cpp [hunk 1]
```diff
@@ -374,6 +374,12 @@ void bringToFront(QWidget* w)
         // activateWindow() (sometimes) helps with keyboard focus on Windows
         if (w->isMinimized()) {
             w->showNormal();
+            // This is a workaround when the main window still minimized to the taskbar (iconified)
+            // due to a (possible) Qt bug or a weird window manager behaviour.
+            if (isObscured(w)) {
```
Comment (jonasschnelli) [inline src/qt/guiutil.cpp]: Does calling `isObscured()` directly after `showNormal()` make sense or should it hit back the the runloop first?

**Comment by hebasto** [comment]
@jonasschnelli 
Thank you for your review and your hint making me investigate the problem more deeper.
The root of problem has been detected: X11 and/or the window manager fail to handle the main window properly when `QSystemTrayIcon::contextMenu()` is a child of the main window.
There is no more ugly workarounds.

Would you mind re-reviewing?

**Comment by jonasschnelli** [comment]
utACK 40d3f19d20040ff1e0b2754fd083d0bb269c2fb5

**Comment by fanquake** [comment]
utACK 40d3f19

File: src/qt/bitcoingui.cpp [hunk 2]
```diff
@@ -224,6 +224,10 @@ BitcoinGUI::~BitcoinGUI()
     settings.setValue("MainWindowGeometry", saveGeometry());
     if(trayIcon) // Hide tray icon, as deleting will let it linger until quit (on Ubuntu)
         trayIcon->hide();
+    if (trayIconMenu) {
```
Comment (laanwj) [inline src/qt/bitcoingui.cpp]: We might want to use a `unique_pointer` and RAII here?

**Comment by laanwj** [comment]
utACK (modulus nit)

@hebasto Thanks a lot for doing a thorough investigation here and not piling on more hacks that seem to do the job.

**Comment by hebasto** [comment]
@laanwj Thank you for your review.

> We might want to use a unique_ptr and RAII here?

Your comment has been addressed. Would you mind re-reviewing?

**Comment by laanwj** [comment]
thank you, utACK a88640e123ca0c00d81719f9a009699c26e85b90

**Pull request merged**